src/scripts/move.js

```diff
@@ -0,0 +1,247 @@
+      .classList.add('field-cell--' + values[randomValue]);
+
+    emptyFields[randomIndex]
+      .innerText = values[randomValue];
+
+  // if its impossible to create a new item:
+  } else {
+    document.querySelector('.message-lose').classList.remove('hidden');
+    document.querySelector('.message-play').classList.add('hidden');
+
+    if (Number(localStorage.getItem('score')) < Number(gameScore.innerText)) {
+      localStorage.setItem('score', gameScore.innerText);
+
+      highestScore.innerText = localStorage.getItem('score');
+    }
+
+    document.body.removeEventListener('keyup', arrowsPlayHandler);
+  }
+}
+
+// Main movement algorithm:
+const arrowsPlayHandler = (e) => {
+  let isMoved = false;
+
+  if (e.key === 'ArrowDown') {
+    // MOVE DOWN:
+    // console.log('DOWN');
+
+    const notEmptyFields = getNotEmptyFields(allCells);
+    const notEmptyFieldsPositions = getNotEmptyFieldsCoords(allCells);
+
+    for (const field of notEmptyFields.reverse()) {
+      const fieldPos = field.id.split('-').slice(1);
+
+      for (const cell of [...allCells].reverse()) {
+        const cellPos = cell.id.split('-').slice(1);
+
+        if (fieldPos[0] < cellPos[0]
+          && fieldPos[1] === cellPos[1]
+          && cell.classList.length === 1
+          && cell.id !== field.id
+        ) {
+          moveCell(field, cell);
+          isMoved = true;
+          break;
+        } else if (fieldPos[0] < cellPos[0]
+          && fieldPos[1] === cellPos[1]
+          && cell.classList.length !== 1
+          && cell.id !== field.id
+          && field.innerText === cell.innerText
+          && checkBetween(fieldPos, cellPos, notEmptyFieldsPositions, 0)
+        ) {
+          margeCells(field, cell);
+          isMoved = true;
+          break;
+        }
+      }
+    }
+
+    if (isMoved || notEmptyFields.length === 16) {
+      createItemInRandomEmptyField();
+    }
+  } else if (e.key === 'ArrowUp') {
+    // MOVE UP:
+    // console.log('UP');
+
+    const notEmptyFields = getNotEmptyFields(allCells);
+    const notEmptyFieldsPositions = getNotEmptyFieldsCoords(allCells);
+
+    for (const field of notEmptyFields) {
+      const fieldPos = field.id.split('-').slice(1);
+
+      for (const cell of [...allCells]) {
+        const cellPos = cell.id.split('-').slice(1);
+
+        if (fieldPos[0] > cellPos[0]
+          && fieldPos[1] === cellPos[1]
+          && cell.classList.length === 1
+          && cell.id !== field.id
+        ) {
+          moveCell(field, cell);
+          isMoved = true;
+          break;
+        } else if (fieldPos[0] > cellPos[0]
+          && fieldPos[1] === cellPos[1]
+          && cell.classList.length !== 1
+          && cell.id !== field.id
+          && field.innerText === cell.innerText
+          && checkBetween(fieldPos, cellPos, notEmptyFieldsPositions, 0, true)
+        ) {
+          margeCells(field, cell);
+          isMoved = true;
+          break;
+        }
+      }
```

Comment:
Let's move logic from this cases to separated functions, it's really hard to read